src/mmrelay/meshtastic_utils.py

```
@@ -3009,61 +3131,46 @@ async def check_connection() -> None:
-                        )
-                    else:
-                        logger.debug(
-                            "Skipping reconnection trigger - already reconnecting"
-                        )
+            except Exception as exc:
+                # Only trigger reconnection if we're not already reconnecting
+                if not reconnecting:
+                    logger.exception(
+                        "%s connection health check failed: %s",
+                        connection_type.capitalize(),
+                        exc,
```

Ruff (0.15.4)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)